packages/core/src/extensions/Comments/CommentsPlugin.ts

```diff
@@ -0,0 +1,297 @@
+import { Node } from "prosemirror-model";
+import { Plugin, PluginKey } from "prosemirror-state";
+import { Decoration, DecorationSet } from "prosemirror-view";
+import { getRelativeSelection, ySyncPluginKey } from "y-prosemirror";
+import { BlockNoteEditor } from "../../editor/BlockNoteEditor.js";
+import { User } from "../../models/User.js";
+import { EventEmitter } from "../../util/EventEmitter.js";
+import { ThreadStore } from "./threadstore/ThreadStore.js";
+import { CommentBody, ThreadData } from "./types.js";
+import { UserStore } from "./userstore/UserStore.js";
+
+const PLUGIN_KEY = new PluginKey(`blocknote-comments`);
+const SET_SELECTED_THREAD_ID = "SET_SELECTED_THREAD_ID";
+
+type CommentsPluginState = {
+  /**
+   * Store the positions of all threads in the document.
+   * this can be used later to implement a floating sidebar
+   */
+  threadPositions: Map<string, { from: number; to: number }>;
```

Comment:
I'm working on some bugs in the suggestion plugin that will be related to this. It isn't safe to store positions like this. Would likely also need to keep around a relative position as well. But, we can figure this out later